### DISCUSSION TITLE
[Android] Annotations Support

### DISCUSSION THREAD

**Discussion by triniwiz** [NativeScript/rfcs#41] [1. Ideas & Discussions]
Introduce support for annotations Vladimir went a bit in-dept [here](https://github.com/NativeScript/android-runtime/issues/1346)

**Comment by farfromrefug** [comment]
Ok so i think it would good to first reference all the decorators we should or already have with a simple example.
Then seeing all of them we could see which one we can merge together with options, and then have a clear version of where we are going

## NativeClass
Simple decorator for extending native class (only one level) used during build time (and removed) to tell the compiler to make an es5 class out of it so that we can use our special sauce 
```ts
@NativeClass
class ExtendedObject extends java.lang.Object {
} 
```
